portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,47 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Optional;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private static final ArrayList<String> COMMENTS = new ArrayList<>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Kevin!</h1>");
+    response.setContentType("application/json");
+    response.getWriter().println(convertCommentsToJson());
+  }
+
+  private String convertCommentsToJson() {
+    Gson gson = new Gson();
+    return gson.toJson(COMMENTS);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<String> comment = getComment(request);
+    if (comment.isPresent()) {
+      COMMENTS.add(comment.get());
+    }
+    response.sendRedirect("./index.html");
+  }
+
+  private Optional<String> getComment(HttpServletRequest request) {
+    String comment = request.getParameter("comment-input-field");
+    if (comment == null || comment.equals("")) {
```

Comment:
good job on using optional 👍

Reply:
I think we should check trailing spaces as well (e.g what happens if the comment is just " " or "<enter>"
hint: use trim()

Reply:
Thank you! I normally use Optional to err on the side of caution. Was thinking about using trim and have included it in the latest commit.